docs/api_reference/flax.linen/layers.rst

```diff
@@ -148,7 +148,7 @@ Recurrent
   Sequential
   Dropout
   SelfAttention
-  MultiHeadDotProductAttention
+  MultiHeadAttention
```

Comment:
maybe we should document both until we finally remove the `MultiHeadDotProductAttention` alias?